### ISSUE TITLE
`pnpm run test --` is not handled as a package script when specified in `jestCommandLine`

### ISSUE THREAD

**Issue by ermik** [jest-community/vscode-jest#974]
### Environment
Jest 29 & PNPM 7

### Prerequisite
- are you able to run jest test from the command line? yes
- how do you run your tests from the command line? yes

### Steps to Reproduce

Set `jestCommandLine` to `pnpm run test --` and use the debug test workflow.

### Expected Behavior

The package script will be correctly delegated to for jest binary invocation.

### Actual Behavior

The invocation will show node invoking the specified string expecting it to be a path to a module:
```
[...] /usr/local/bin/node ./pnpm run test
```
causing "Cannot find module" error. This works with `npm run test` because usage of a "magic string" (`npm`) changes the debug invocation to refer to the package manager instead of the local `node` binary.

**Comment by connectdotz** [comment]
That's true, want to work on a PR? should be pretty straightforward: https://github.com/jest-community/vscode-jest/blob/3deb1af4586aaf6baf2ae30cdced34e77575a436/src/DebugConfigurationProvider.ts#L157

**Comment by ermik** [comment]
Thanks for asking, I might. Will post here if I pick this up. 💟

**Comment by orpheus6678** [comment]
I have faced the weirdest issue that might be related. I was using a Windows machine and in a Next.js project of mine I have Jest 29.7 configured manually, exactly as depicted in [this article](https://nextjs.org/docs/app/building-your-application/testing/jest#manual-setup). If there was any key difference from a typical Next.js project, it is that I was using pnpm. I took care to set `jestCommandLine` to `pnpm run test --` (package.json/scripts/test was set to "jest" of course) but if I did that, the command line built upon running from the extension is like

`jest "--" "--testLocationInResults" "--json" "--useStderr" ...`

and then Jest wouldnt find any of my tests and exit with error (`testRegex` would get 0 matches). I found out that the "--" in jest invocation was causing all this, what I did not expect was that setting `jestCommandLine` to `pnpm test --` would fix this issue. It makes no sense, but then maybe [this](https://pnpm.io/cli/run#options) was the cause.

**Comment by icy0307** [comment]
https://github.com/jest-community/vscode-jest/blob/c93ec7a278ec84c2ac909d16ed155caccaeca933/setup-wizard.md?plain=1#L127-L133
According to setup-wizard "How is debug config generated" section, jest plugin updates `runtimeExecutable` property if the command is `npm` or `yarn` , otherwise updates `program`.
pnpm is not one of it , so it  updates `program` to ` "program": "[project root]/node_modules/.bin/jest"`.
```json
{
    "configurations": [
        {
            "type": "node",
            "name": "vscode-jest-tests.v2..leetcode",
            "request": "launch",
            "args": [
                "--runInBand",
                "--watchAll=false",
                "--testNamePattern",
                "${jest.testNamePattern}",
                "--runTestsByPath",
                "${jest.testFile}"
            ],
            "cwd": "/Users/bubu/.leetcode",
            "console": "integratedTerminal",
            "internalConsoleOptions": "neverOpen",
            "disableOptimisticBPs": true,
            "program": "[project root]/node_modules/.bin/jest" // PNPM REWRITES IT TO SHELL BY DEFAULT
        }
    ]
}
```
However , the core problem is that **pnpm creates the Jest binary file (node_modules/.bin/jest) as a shell script** rather than a Node.js script, which cause issue like #868 . and this is also why #1046 would not work.
```
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
          ^^^^^^^

SyntaxError: missing ) after argument list
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1033:15)
    at Module._compile (node:internal/modules/cjs/loader:1069:27)
    at Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Module._load (node:internal/modules/cjs/loader:827:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47
```
User's own jestCommandLine might also not be node js as well, so the program field is not the answer. (scripts in package.json could also be shell scripts)
The alternatives' are
1. make `runtimeExecutable ` in `package.json` `sh` (But this probably would not work on windows, we need find out how to work on window)
2. make `runtimeExecutable `  pnpm 
Pnpm do have settings like [prefer-symlinked-executables](https://pnpm.io/npmrc#prefer-symlinked-executables), but it is not default and might not be used on windows. 
so it would be nice to make this plugin support pnpm and other future package managers.
